### PR TITLE
ARTEMIS-562 Use 'to' field if sender target is null

### PR DESCRIPTION
**Pull request by gaohoward**

### CODE REVIEW

**Comment by gaohoward** [comment]
Based on Time's patch.

File: artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPSessionCallback.java [hunk 1]
```diff
@@ -429,6 +442,12 @@ public void offerProducerCredit(final String address,
                                    final int threshold,
                                    final Receiver receiver) {
       try {
+         // TODO - Do something smart here with the paging bits to figure out the credit can be sent.
```
Comment (clebertsuconic) [inline artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPSessionCallback.java]: I don't think there's anything to be done with paging. If this is an anonymous producer, there's nothing to be done with paging...

Either this is the proper implementation, hence you would need to remove the TODO, or this is something else broken.

I believe you should just remove the TODO here

Comment (gaohoward) [inline artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPSessionCallback.java]: Thanks clebert. I'll remove it.

**Comment by gaohoward** [comment]
Just checked the failure not related to this PR, and it's probably not a real issue. Pass always on local.

**Pull request closed without merging**